lib/core/src/display/scene/htmlrenderer.rs

```diff
@@ -0,0 +1,104 @@
+        scene
+            .div
+            .element
+            .style()
+            .set_property("perspective", &format!("{}px", fov))
+            .expect("set perspective");
+        scene
+            .camera
+            .element
+            .style()
+            .set_property(
+                "transform",
+                &format!(
+                    "translateZ({}px)
+             matrix3d({}, {}, {}, {},
+                      {}, {}, {}, {},
+                      {}, {}, {}, {},
+                      {}, {}, {}, {})
+             translate({}px, {}px)",
+                    fov,
+                    eps(transform[0]),
+                    eps(-transform[1]),
+                    eps(transform[2]),
+                    eps(transform[3]),
+                    eps(transform[4]),
+                    eps(-transform[5]),
+                    eps(transform[6]),
+                    eps(transform[7]),
+                    eps(transform[8]),
+                    eps(-transform[9]),
+                    eps(transform[10]),
+                    eps(transform[11]),
+                    eps(transform[12]),
+                    eps(-transform[13]),
+                    eps(transform[14]),
+                    eps(transform[15]),
+                    view_width / 2.0,
+                    view_height / 2.0
+                ),
+            )
+            .expect("set camera transform");
+
+        for object in &scene.objects.items {
+            match object {
+                Some(object) => {
+                    let transform = object.transform.to_homogeneous();
+                    object
+                        .element
+                        .style()
+                        .set_property(
+                            "transform",
+                            &format!(
+                                "translate(-50%, -50%)
+                         matrix3d({}, {}, {}, {},
+                                  {}, {}, {}, {},
+                                  {}, {}, {}, {},
+                                  {}, {}, {}, {})",
+                                eps(transform[0]),
+                                eps(transform[1]),
+                                eps(transform[2]),
+                                eps(transform[3]),
+                                eps(-transform[4]),
+                                eps(-transform[5]),
+                                eps(-transform[6]),
+                                eps(-transform[7]),
+                                eps(transform[8]),
+                                eps(transform[9]),
+                                eps(transform[10]),
+                                eps(transform[11]),
+                                eps(transform[12]),
+                                eps(transform[13]),
+                                eps(transform[14]),
+                                eps(transform[15])
+                            ),
+                        )
```

Comment:
This looks like copy-paste from previous code. Definitely to be refactored.

Reply:
The negation is different. This formatted code is definitely hard to read.